<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!--baselink--> 
    <!--node pkg="tf" type="static_transform_publisher" name="rs16_to_baselink_broadcaster" args="-0.314 -0.0155 -0.261 -0.0117 -0.00218 1.0 -0.00688 rs16 baselink 10"/-->
    <!--node pkg="tf" type="static_transform_publisher" name="rs32_to_baselink_broadcaster" args="0 0 0 0 0 0 1 baselink rs32 10"/-->
    <node pkg="tf" type="static_transform_publisher" name="rs16_to_rs32_broadcaster" args="-0.314 -0.0155 -0.261 -0.0117 -0.00218 1.0 -0.00688 rs16 rs32 10"/>
    <!-- Camera topic and weights, config and classes files -->
    <arg name="ros_bag_file"	                default="/home/ps/hxw_projects/OpenPCDet_working/catkin_ws/src/hf3d_ros/bag/indoor1.bag"/>

    <node pkg="rosbag" type="play" name="player" output="log" args="-l $(arg ros_bag_file)" />

    <!-- Camera topic and weights, config and classes files -->
    <arg name="rs16_topic"	                default="/ns1/rslidar_points"/>
    <arg name="rs32_topic"	            default="/ns2/rslidar_points"/>

    <!-- Published topics -->
    <arg name="rs16_to_rs32_topic"      default="/ns3/pc_rs16_under_rs32"/>

    <!-- Node -->
    <node name="detector_manager" pkg="yolov3_pytorch_ros" type="convert_pc_rs16_to_rs32.py" output="screen" respawn="true">
        <param name="rs16_topic"               value="$(arg rs16_topic)"/>
        <param name="rs32_topic"              value="$(arg rs32_topic)"/>
        <param name="rs16_to_rs32_topic"               value="$(arg rs16_to_rs32_topic)"/>
    </node>
    <node type="rviz" name="rviz" pkg="rviz"/>
</launch>